### DISCUSSION TITLE
Component Watch Survives if the watched property is a store Property

### DISCUSSION THREAD

**Discussion by T0miii** [vuejs/pinia#1328] [Help and Questions] [answered]
### Reproduction

In the description

### Steps to reproduce the bug

Have a Component mounted witch is watching a property of the store, via storeToRefs, recreate the component by changeing its :key property, see the component console.log unmounted and mounted again, mutate the state of the store property, see the watch triggering twice.

My App.vue:
`  <div>
    <el-button @click="() => val++">RESET</el-button>
    <Filter :key="val"></Filter>
  </div>
`

My Component:

```
<script setup lang="ts">

import {onMounted, onUnmounted, watch} from 'vue'
import {storeToRefs} from 'pinia'
import {useTestStore} from '@/views/bata/testStore'
const testStore = useTestStore()
const {testArray} = storeToRefs(testStore)

const addTo = () => {
  testArray.value.push({id: '1'})
}

const id = Math.random() *100
watch(testArray, (array) => {
  console.log('Watch Widget List', array, id)
}, {deep: true})

onMounted(() => {
  console.log('Component Mounted', id)
})

onUnmounted(() => {
  console.log('Component unmounted', id)
})

</script>
<template>
  <div style="border: 1px solid; width: 200px">
    <h1>My Component</h1>
    <el-button @click="addTo">Add to array</el-button>
  </div>
</template>
```
As you can see the component is mounting with a controll ID, and when i mutate the state the watch is triggering as expected, after i recreate the component with changing its key property, the component unmounts wth the ID and mounts with the new ID. After mutating the state the old watch is still triggering:

![image](https://user-images.githubusercontent.com/10532141/171368383-662623ad-5c6c-4f78-b2a6-d9eec2f2ba43.png)

### Expected behavior

Expect the watch to be destoryed when the component unmounts.

### Actual behavior

The watch is triggering twice after the component is unmounted

### Additional information

I also want to be honest and say that i could no recreate this scenario in a different webpack 5 project, but then again i removed everything from my App beside the App.vue thats the entry point and the Test Component.

On a side note, if i manually const unwatch = watch() .....  unwatch() in the onUnmounted it works as expected, but this is not how it suposed to be i think.

**Comment by posva** [comment, marked as answer]
I couldn't reproduce it either. Maybe you have outdated versions? Let me know if you manage to reproduce but also please don't open issues without a reproduction 🙏

**Reply by T0miii** [reply]
i will try to remove everything from our project and push it to git. It is a really strange behavior.

**Reply by T0miii** [reply]
As we prepared the project and last step removed the node_modules and package_lock (as we wanted to publish the example) and reinstalled everything, this unmounted watch behavior stoped.

So it seems there was really a version problem? 

Anyhow sorry for posting a bug, should have put it right away into the discussion section.
